trieste/acquisition/rule.py

```diff
@@ -1103,6 +1115,380 @@ def state_func(
         return state_func
 
 
+class UpdateableSearchSpace(SearchSpace):
+    """A search space that can be updated."""
+
+    @abstractmethod
+    def reinitialise(
+        self,
+        models: Optional[Mapping[Tag, ProbabilisticModelType]] = None,
+        datasets: Optional[Mapping[Tag, Dataset]] = None,
+    ) -> None:
+        """Reinitialise the search space using the given models and datasets."""
+        ...
+
+    @abstractmethod
+    def update(
+        self,
+        models: Optional[Mapping[Tag, ProbabilisticModelType]] = None,
+        datasets: Optional[Mapping[Tag, Dataset]] = None,
+    ) -> None:
+        """Update the search space using the given models and datasets."""
+        ...
+
+    def get_single_model_and_dataset(
+        self,
+        models: Optional[Mapping[Tag, ProbabilisticModelType]] = None,
+        datasets: Optional[Mapping[Tag, Dataset]] = None,
+    ) -> Tuple[Optional[ProbabilisticModelType], Optional[Dataset]]:
+        """Get a single model and dataset from the given arguments."""
+        if models is None:
+            model = None
+        elif OBJECTIVE in models.keys():
+            model = models[OBJECTIVE]
+        else:
+            raise ValueError(f"""if models is provided, it must contain the key {OBJECTIVE}""")
```

Comment:
This is a helper method for the simple default case, to make developers' life easier. If a different model/dataset is required, than the expectations is to use the full datasets/models dictionaries directly instead of calling this.